tests/storage-disks-vm

```diff
@@ -170,6 +170,96 @@ if [ "${DEVICE_HOTPLUG}" = "1" ]; then
       ! lxc exec v2 -- mountpoint /mnt/bar2 || false # Just tests for mountpoint disconnected, not unmount.
   fi
 fi
+
+echo "==> Checking additional disk device support"
+lxc stop -f v2
+if [ "${LONG_DEVICE_NAME}" = "notSoLongName" ]; then
+  lxc config device remove v2 "${LONG_DEVICE_NAME}"
+fi
+
+# Add disks
+lxc config device add v2 dir1rw disk source="${testRoot}/allowed1" path="/srv/rw"
+lxc config device add v2 dir1ro disk source="${testRoot}/allowed1" path="/srv/ro" readonly=true
+lxc config device add v2 block1ro disk source="${testRoot}/allowed1/lxd-block-test" readonly=true
+lxc config device add v2 block1rw disk source="${testRoot}/allowed1/lxd-block-test"
+lxc start v2
+waitInstanceReady v2
+
+lxc exec v2 -- mount | grep -F /srv/
+
+echo "==> Testing VM lxd-agent drive mounts"
+# Check there is only 1 mount for each directory disk and that it is mounted with the appropriate options.
+[ "$(lxc exec v2 -- mount | grep -cF '/srv/rw type')" = "1" ]
+[ "$(lxc exec v2 -- mount | grep -cF '/srv/ro type')" = "1" ]
+
+# RW disks should use virtio-fs when used with the snap.
+lxc exec v2 -- mount | grep 'lxd_dir1rw on /srv/rw type virtiofs (rw,relatime)'
+
+# RO disks should use virtio-fs when used with the snap but be mounted readonly.
+lxc exec v2 -- mount | grep 'lxd_dir1ro on /srv/ro type virtiofs (ro,relatime)'
+
+# Check UID/GID are correct.
+[ "$(lxc exec v2 -- stat -c '%u:%g' /srv/rw)" = '0:0' ]
```

Comment:
@simondeziel The tests are failing here because the user and group come out as 65534 (nobody) and not 0 (root). I am unsure how to handle this since changing the device source directory would break the tests due to profile restrictions and neither creating the directory with `sudo` nor changing the tests to expect `65534:65534` seem like appropriate solutions.